lib/elixir/lib/kernel.ex

```diff
@@ -8,33 +8,26 @@ import :elixir_bootstrap
 
 defmodule Kernel do
   @moduledoc """
-  This module is the entry point of the Elixir programming language.
-
-  It provides the default functions and macros Elixir imports into your
-  environment. Those can be directly invoked in your code without the
-  module name:
-
+  `Kernel` is Elixir's set of core functionality, mainly consisting of:
+  
+    * syntactic sugar - for more readable and intuitive code
+    * [macros](https://elixir-lang.org/getting-started/meta/macros.html) - for defining new functionality
+    * [guard](guards.html) checks - for more powerful function definitions
+    
+  `Kernel` is automatically imported into all `.ex` and `.exs` files,
+  meaning that you can use its functions and macros in any file or `IEx` 
+  without using the module name:
+  
       iex> is_number(13)
       true
-
-  These functions and macros can be skipped or cherry-picked via the
-  `Kernel.SpecialForms.import/2` macro. For instance, if you want to tell
-  Elixir not to import the `if/2` macro, you can do:
-
-      import Kernel, except: [if: 2]
+      
+  For optimizations, you can tell Elixir not to import a given function or macro:
```

Comment:
I'm not sure I would call it optimisations - the primary use case I found for this in the wild is defining a `send/2` function in your module that won'r conflict with `Kernel.send/2` - it usually has nothing to do with performance.

Reply:
#TIL ❤️